### PR TITLE
(#24893) magic_enum/0.9.6: fixes install dir

### LINKED ISSUE #24893: [package] magic_enum/0.9.6: Include path format
### Description

**Observed behavior**

The include path of the conan package `magic_enum/0.9.6` is 
```
#include <magic_enum/magic_enum.hpp>
```

**Expected behavior**

According to the CMake install target and following specification of magic_enum maintainer in 
https://github.com/Neargye/magic_enum/issues/310#issuecomment-1808234766, the include path should be without prefix:

```
#include <magic_enum.hpp>
```

The problem is also illustrated by the [test-package](https://github.com/conan-io/conan-center-index/blob/f3c3ac936f3ac485c5445b038a3b897607a3d2c1/recipes/magic_enum/all/test_package/test_package.cpp) which includes the following workaround for the issue:

```
#ifdef MAGIC_ENUM_INCLUDE_WITH_FOLDER
#  include <magic_enum/magic_enum.hpp>
#else
#  include <magic_enum.hpp>
#endif
```

### Package and Environment Details

* Package Name/Version: **magic_enum/0.9.6**

### Conan profile

Host profile:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=12
os=Linux

Build profile:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=12
os=Linux

### Steps to reproduce

Run CMake install of `magic_enum`
```
git clone https://github.com/Neargye/magic_enum.git
cd magic_enum
mkdir build && cd build
mkdir magic_enum_package
cmake -DCMAKE_INSTALL_PREFIX=magic_enum_package ..
make
make install 
```

verify the installation without a directory `magic_enum` in output of `make install`:
```
[  4%] Built target example
[  9%] Built target enum_flag_example
[ 13%] Built target example_containers_array
[ 18%] Built target example_containers_bitset
[ 22%] Built target example_containers_set
[ 27%] Built target example_custom_name
[ 31%] Built target example_switch
[ 36%] Built target test-cpp17
[ 40%] Built target test_flags-cpp17
[ 45%] Built target test_aliases-cpp17
[ 50%] Built target test_containers-cpp17
[ 54%] Built target test_wchar_t-cpp17
[ 59%] Built target test-cpp20
[ 63%] Built target test_flags-cpp20
[ 68%] Built target test_aliases-cpp20
[ 72%] Built target test_containers-cpp20
[ 77%] Built target test_wchar_t-cpp20
[ 81%] Built target test-cpp23
[ 86%] Built target test_flags-cpp23
[ 90%] Built target test_aliases-cpp23
[ 95%] Built target test_containers-cpp23
[100%] Built target test_wchar_t-cpp23
Install the project...
-- Install configuration: ""
-- Up-to-date: /home/nine/src/github.com/Neargye/magic_enum/build/magic_enum_package/include/magic_enum.hpp
-- Up-to-date: /home/nine/src/github.com/Neargye/magic_enum/build/magic_enum_package/include/magic_enum_all.hpp
-- Up-to-date: /home/nine/src/github.com/Neargye/magic_enum/build/magic_enum_package/include/magic_enum_containers.hpp
-- Up-to-date: /home/nine/src/github.com/Neargye/magic_enum/build/magic_enum_package/include/magic_enum_flags.hpp
-- Up-to-date: /home/nine/src/github.com/Neargye/magic_enum/build/magic_enum_package/include/magic_enum_format.hpp
-- Up-to-date: /home/nine/src/github.com/Neargye/magic_enum/build/magic_enum_package/include/magic_enum_fuse.hpp
-- Up-to-date: /home/nine/src/github.com/Neargye/magic_enum/build/magic_enum_package/include/magic_enum_iostream.hpp
-- Up-to-date: /home/nine/src/github.com/Neargye/magic_enum/build/magic_enum_package/include/magic_enum_switch.hpp
-- Up-to-date: /home/nine/src/github.com/Neargye/magic_enum/build/magic_enum_package/include/magic_enum_utility.hpp
-- Up-to-date: /home/nine/src/github.com/Neargye/magic_enum/build/magic_enum_package/lib/cmake/magic_enum/magic_enumConfig.cmake
-- Up-to-date: /home/nine/src/github.com/Neargye/magic_enum/build/magic_enum_package/lib/cmake/magic_enum/magic_enumConfigVersion.cmake
-- Up-to-date: /home/nine/src/github.com/Neargye/magic_enum/build/magic_enum_package/lib/pkgconfig/magic_enum.pc
-- Up-to-date: /home/nine/src/github.com/Neargye/magic_enum/build/magic_enum_package/share/magic_enum/package.xml

```

### Logs

logs were attached above

### PR DESCRIPTION
**Pull request by nine**
### Summary
Changes to recipe:  **magic_enum/0.9.6**

#### Motivation
This MR is intended to fix #24893

#### Details
This MR is intended to fix #24893

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

File: recipes/magic_enum/all/conanfile.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def build(self):
         pass
 
     def package(self):
-        copy(self, "*", src=os.path.join(self.source_folder, "include"), dst=os.path.join(self.package_folder, "include"))
+        copy(self, "*", src=os.path.join(self.source_folder, "include/magic_enum"), dst=os.path.join(self.package_folder, "include"))
```
Comment (SpaceIm) [inline recipes/magic_enum/all/conanfile.py:63]:
```suggestion
        if Version(self.version) < "0.9.4":
            copy(self, "*", src=os.path.join(self.source_folder, "include"), dst=os.path.join(self.package_folder, "include"))
        else:
            copy(self, "*", src=os.path.join(self.source_folder, "include", "magic_enum"), dst=os.path.join(self.package_folder, "include"))
```

Comment (nine) [reply recipes/magic_enum/all/conanfile.py]: @SpaceIm Thanks for the feedback regarding support of all package versions of `magic_enum`.

In order to achieve that I've changed the `package()` method to use the CMake install target.

**Comment by jcar87** [comment]
Given that this was already out in the wild, I would probably err on the side of caution and at least for the published versions that had `magic_enum/magic_enum.hpp` I would support both (that is, copy the files in the package in two places). Otherwise we will break anyone who might have been using it that way. There should (hopefully) be no downsides in this.

File: recipes/magic_enum/all/conanfile.py [hunk 2]
```diff
@@ -39,7 +39,7 @@ def _compilers_minimum_version(self):
         }
 
     def layout(self):
-        basic_layout(self, src_folder="src")
+        cmake_layout(self)
```
Comment (SpaceIm) [inline recipes/magic_enum/all/conanfile.py:42]:
```suggestion
        cmake_layout(self, src_folder="src")
```

**Comment by toge** [comment]
@nine @SpaceIm 
Sorry for taking up your time with my mistake.
Thank you for creating the PR that makes it the correct recipe.

File: recipes/magic_enum/all/conanfile.py (hunk 2 above)
Comment (nine) [reply recipes/magic_enum/all/conanfile.py]: Thank you, added change in [00457e3](https://github.com/conan-io/conan-center-index/pull/24894/commits/00457e341a1bc4c1be52db39cdb8e091a5461737).

**Comment by nine** [comment]
@jcar87 according to your suggestion in https://github.com/conan-io/conan-center-index/pull/24894#issuecomment-2283323701 I've added a copy of the includes in the legacy directory for package versions from 0.9.4 to 0.9.6 in https://github.com/conan-io/conan-center-index/pull/24894/commits/ecdbfa032eca40c3b10725831d85a31d2d639f44.

**Review by AbrilRBS — APPROVED** [review]
Thanks!

**Review by ErniGH — APPROVED** [review]
LGTM

**Pull request merged**